### PR TITLE
MINIFI-131: Provenance Support

### PR DESCRIPTION
**Pull request by benqiu2016**

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: libminifi/include/Provenance.h [hunk 1]
```diff
@@ -0,0 +1,902 @@
...
+	{
+		event->Serialize(this);
+	}
+	//! Remove event
+	void removeEvent(ProvenanceEventRecord *event)
+	{
+		Delete(event->getEventId());
+	}
+	//! destory
+	void destory()
```
Comment (apiri) [inline libminifi/include/Provenance.h]: should adjust this to destroy

**Comment by apiri** [comment]
@benqiu2016 also, we should update .travis.yml with the leveldb dependency.

I verified this corrects the CI build on a branch at: https://github.com/apiri/nifi-minifi-cpp/blob/travis-leveldb/.travis.yml#L38

**Comment by benqiu2016** [comment]
@apiri 
address above review comments 
Fix the GetFile regex bug with this PR also. https://issues.apache.org/jira/browse/MINIFI-113

**Comment by apiri** [comment]
Hey @benqiu2016,

Changes look good and will merge.  Moving forward, just as a note, we should try to keep PRs on a one-to-one basis with JIRA issues to provide easier tracking.

Thanks for taking this on, looks great!

**Pull request closed without merging**